tools/ci_build/github/azure-pipelines/templates/win-ci-vs-2019.yml

```diff
@@ -119,14 +120,35 @@ jobs:
     workingDirectory: '$(Build.BinariesDirectory)'
     displayName: 'Install python modules'
 
+  - powershell: |
+      if ([string]::IsNullOrEmpty((Get-Command ccache -errorAction SilentlyContinue)))
+      {
+        choco install ccache -y --version 4.7.4
+        $ccache_path = (Get-Command ccache).Source
+        $ccache_parent_dir = (Split-Path -parent $ccache_path)
+        Copy-Item "C:\ProgramData\chocolatey\lib\ccache\tools\ccache-4.7.4-windows-x86_64\ccache.exe" -Destination "C:\ProgramData\chocolatey\bin\cl.exe"
+        Get-ChildItem $ccache_parent_dir
+        ccache --version
+      }
+    displayName: Install ccache and update PATH to use linked versions of gcc, cc, etc
+
+  - task: Cache@2
+    inputs:
+      key: 'ccache | "$(System.StageName)" | "$(Build.SourceBranch)" | "$(Build.SourceVersion)" '
+      path: $(CCACHE_DIR)
+      restoreKeys: |
+        ccache | "$(System.StageName)" | "$(Build.SourceBranch)"
+        ccache | "$(System.StageName)"
+        ccache |
+    displayName: Cache Task
+
   - ${{ if or(eq(parameters.RunOnnxRuntimeTests, true), eq(parameters.GenerateDocumentation, true)) }}:
       - task: PowerShell@2
         displayName: 'Install ONNX'
         inputs:
           filePath: '$(Build.SourcesDirectory)/tools/ci_build/github/windows/install_third_party_deps.ps1'
           workingDirectory: '$(Build.BinariesDirectory)'
-          arguments: -cpu_arch ${{ parameters.buildArch }} -install_prefix $(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\installed -build_config ${{ parameters.BuildConfig }}
-
+          arguments: -cpu_arch ${{ parameters.buildArch }} -install_prefix $(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\installed -build_config ${{ parameters.BuildConfig }} -use_cache
```

Comment:
If this config is used in any packaging pipeline builds we probably want to make enabling ccache optional. Typically you want those to be completely clean even if that means they take longer.

Reply:
@skottmckay 
Thank your suggestion. It's been improved.
By default, the cache step is skipped.
![image](https://user-images.githubusercontent.com/16190118/210691331-1b90b02f-bf43-46a6-9ab4-5730bfd7aeff.png)